### ISSUE TITLE
`pandas` `infer_objects` type stub missing `copy` arg

### ISSUE THREAD

**Issue by janosh** [astral-sh/ty#1076]
### Summary

both calls to `infer_objects` on `pandas` `DataFrame` and `Series` trigger

> Argument `copy` does not match any known parameter of bound method `infer_objects` (https://ty.dev/rules#unknown-argument)

```py
import pandas as pd

df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6], "C": [7, 8, 9]})

df.infer_objects(copy=False)
df.A.infer_objects(copy=False)
```

docs on the `copy` parameter:

- https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.infer_objects.html
- https://pandas.pydata.org/docs/reference/api/pandas.Series.infer_objects.html

i looked over at https://github.com/search?q=repo%3Apython%2Ftypeshed%20infer_objects&type=code but found no mention of `infer_objects` so not sure where this incorrect type stub is coming from

**Comment by AlexWaygood** [comment]
Thanks for the report!

> i looked over at [github.com/search?q=repo%3Apython%2Ftypeshed%20infer_objects&type=code](https://github.com/search?q=repo%3Apython%2Ftypeshed%20infer_objects&type=code) but found no mention of `infer_objects` so not sure where this incorrect type stub is coming from

Typeshed only provides stubs for objects defined in the standard library. (Well, it has stubs for other libraries, but ty doesn't vendor those out of the box -- you have to `(uv) pip install` (or the Conda equivalent) those stubs packages into a Python environment.) The reason why typeshed's stubs were relevant in https://github.com/astral-sh/ty/issues/1048 was that pandas's type annotations indicated that the `Dataframe.attrs` attribute is a dictionary, so ty then looked at the stubs for the `dict` class and its methods in typeshed to figure out what the signature of `dict.update()` should be.

Here, it looks like the `infer_object()` method is directly defined on `pandas.DataFrame`, so I'd expect ty to look either in the `pandas` library or the `pandas-stubs` library for information on the signature of that method. (In general, it'll be much easier to debug issues with `pandas` if you could let us know whether you have `pandas-stubs` installed or not! If it is installed into your environment, ty will look at that stubs package for information about pandas; if not, it'll look at the `pandas` library itself for information about the classes, functions and methods in `pandas`.)

**Comment by AlexWaygood** [comment]
I can reproduce the error if I have `pandas-stubs` installed but not if I only have `pandas` installed, so I'm guessing you have `pandas-stubs` installed into your environment!

**Comment by janosh** [comment]
thanks a lot @AlexWaygood for troubleshooting this. you're exactly right:

```
$ pip show pandas-stubs
Using Python 3.13.0 environment at: /Users/janosh/.venv/py313
Name: pandas-stubs
Version: 2.3.0.250703
Requires: numpy, types-pytz
Required-by:
```

required by nothing so what triggered its installation initially seems to have been uninstalled or no longer depends on it.

i'll report this over at https://github.com/pandas-dev/pandas-stubs

**Comment by AlexWaygood** [comment]
> thanks a lot [@AlexWaygood](https://github.com/AlexWaygood) troubleshooting this. you're exactly right:

No problem!

I think it was harder than it should be here to figure out where ty was getting its type information from -- I'm working on a patch now to improve our diagnostics here

**Comment by AlexWaygood** [comment]
See
- https://github.com/astral-sh/ruff/pull/20022
